### ISSUE TITLE
Next Player Cannot Go

### ISSUE THREAD

**Issue by joshuahaertel** [thyer/CS340winter2016#577]
Possibly linked to Bug 564, but after two players bought two cities the next player to go had a broken client. The rest of the clients were fine.

![screen shot 2016-03-25 at 5 06 33 pm](https://cloud.githubusercontent.com/assets/16691565/14055978/00aa9656-f2ac-11e5-9d49-4eeddb4dda05.png)

They are allowed to do somethings, assumedly invalidly:
![screen shot 2016-03-25 at 5 08 55 pm](https://cloud.githubusercontent.com/assets/16691565/14056011/52c2cd0a-f2ac-11e5-8ed5-c90508bce484.png)

But this did not appear on other clients.

After restarting the client, the game is normal again.

**Comment by joshuahaertel** [comment]
This happened again without actually doing anything. What I did this time was:
Start server
Open four clients
Log in as Sam, then Brooke, then Pete all with their original colors. The fourth one I logged in as Sam to double check that he had the right color selection. I closed it then opened it again and logged in as Mark with his original color. All I did was roll and finish my turn. Swagger was up from the previous time I used the server, however it was not used. The server produced no output.

**Comment by joshuahaertel** [comment]
Closing the clients and then logging in again initially worked, however after Brooke rolled and ended her turn, the other clients broke again.

**Comment by joshuahaertel** [comment]
Interestingly enough, this happens when I login players in reverse order. I can see this being a state problem from phase 2. When I log players in as soon as it is their turn, I am fine.

**Comment by zlohner** [comment]
Which client are you using?

**Comment by joshuahaertel** [comment]
I just followed the same procedure as before and got similar results. This time, when I logged in as Sam with the color yellow his client was broken. The server terminal doesn't output anything so I'm uncertain about the nature of the error. The only other thing I can think of right now is that I checked your client with the demo-server for a previous bug before I ran your server. I do not know why that would affect it because I closed the client in between switching servers, but hopefully this additional information helps you find the source of this bug.
